### PR TITLE
Added configurable block reward and recipient for IBFT2

### PR DESCRIPTION
**Pull request by rain-on**
Allow IBFT2 to deliver block rewards (of a configurable value), and transaction fees to a configured address.

### CODE REVIEW

File: consensus/ibft/src/main/java/org/hyperledger/besu/consensus/ibft/IbftProtocolSchedule.java [hunk 1]
```diff
@@ -60,16 +59,24 @@ public static ProtocolSchedule create(final GenesisConfigOptions config) {
...
+
+    builder
+        .blockHeaderValidatorBuilder(ibftBlockHeaderValidator(ibftConfig.getBlockPeriodSeconds()))
+        .ommerHeaderValidatorBuilder(ibftBlockHeaderValidator(ibftConfig.getBlockPeriodSeconds()))
         .blockBodyValidatorBuilder(MainnetBlockBodyValidator::new)
         .blockValidatorBuilder(MainnetBlockValidator::new)
         .blockImporterBuilder(MainnetBlockImporter::new)
         .difficultyCalculator((time, parent, protocolContext) -> BigInteger.ONE)
-        .blockReward(Wei.ZERO)
+        .blockReward(Wei.of(ibftConfig.getBlocKReward()))
```
[the remark this answers is not in the document]
Comment (rain-on) [reply consensus/ibft/src/main/java/org/hyperledger/besu/consensus/ibft/IbftProtocolSchedule.java]: done

[the remark this answers is not in the document]
Comment (rain-on) [reply consensus/ibft/src/main/java/org/hyperledger/besu/consensus/ibft/IbftProtocolSchedule.java]: Not sure - all other numbers in this section are decimal ... so tempted to stick with that? There doesn't seem to be much of a definitive approach...

File: acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/ibft2/Ibft2BlockRewardPaymentAcceptanceTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.Optional;
+
+import org.junit.Test;
+
+public class Ibft2BlockRewardPaymentAcceptanceTest extends AcceptanceTestBase {
+
+  @Test
+  public void validatorsArePaidBlockReward() throws IOException {
```
Comment (jframe) [inline acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/ibft2/Ibft2BlockRewardPaymentAcceptanceTest.java:34]: Do we need a test with multiple validators receiving a fixed reward? Though that get complicated so might not be worth it.

File: config/src/main/java/org/hyperledger/besu/config/IbftConfigOptions.java [hunk 3]
```diff
@@ -75,6 +77,14 @@ public int getFutureMessagesMaxDistance() {
         ibftConfigRoot, "futuremessagesmaxdistance", DEFAULT_FUTURE_MESSAGES_MAX_DISTANCE);
   }
 
+  public Optional<String> getMiningBeneficiary() {
+    return JsonUtil.getString(ibftConfigRoot, "miningbeneficiary");
+  }
+
+  public long getBlockRewardEth() {
+    return JsonUtil.getLong(ibftConfigRoot, "blockrewardeth", DEFAULT_IBFT_BLOCK_REWARD_ETH);
```
Comment (jframe) [inline config/src/main/java/org/hyperledger/besu/config/IbftConfigOptions.java:85]: Don't think this need eth as a suffix, no other fields do that. Should be clear enough that it's eth.

File: acceptance-tests/tests/src/test/resources/ibft/ibft.json [hunk 4]
```diff
@@ -10,7 +10,8 @@
     "ibft2": {
       "blockperiodseconds": 1,
       "epochlength": 30000,
-      "requesttimeoutseconds": 5
+      "requesttimeoutseconds": 5,
+      "blockrewardeth": 5
```
Comment (jframe) [inline acceptance-tests/tests/src/test/resources/ibft/ibft.json:14]: Rather the default was left as 0 as what most people will use and just modify the blockreward for the block reward AT

File: acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/ibft2/Ibft2BlockRewardPaymentAcceptanceTest.java (hunk 2 above)
Comment (rain-on) [reply acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/ibft2/Ibft2BlockRewardPaymentAcceptanceTest.java:34]: Yeah, it got scary when you start saying "how do you know"? I.e. it depends on rounds, and blocks...
maybe you just wait until 2 validators have been paid (something ... anything) - no API exists in the DSL to do that, but sure we can make it work

File: config/src/main/java/org/hyperledger/besu/config/IbftConfigOptions.java (hunk 3 above)
Comment (rain-on) [reply config/src/main/java/org/hyperledger/besu/config/IbftConfigOptions.java:85]: done

File: acceptance-tests/tests/src/test/resources/ibft/ibft.json (hunk 4 above)
Comment (rain-on) [reply acceptance-tests/tests/src/test/resources/ibft/ibft.json:14]: Yeah agreed - but it's REALLY hard to do - given no existing tests prove that the validator's not paid, I figured this was ok?

File: config/src/test/java/org/hyperledger/besu/config/JsonGenesisConfigOptionsTest.java [hunk 5]
```diff
@@ -116,4 +116,40 @@ public void configWithAnIbftWithNoValidatorsListedIsValid() {
     assertThat(configOptions.getTransitions().getIbftForks().get(1).getValidators().get().size())
         .isEqualTo(1);
   }
+
+  @Test
+  public void configWithValidIbftBlockRewardIsParsable() {
```
Comment (jframe) [inline config/src/test/java/org/hyperledger/besu/config/JsonGenesisConfigOptionsTest.java:121]: Can you also add a test for block rewards with a decimal number

Comment (rain-on) [reply config/src/test/java/org/hyperledger/besu/config/JsonGenesisConfigOptionsTest.java:121]: yeah sure.

**Pull request merged**